### DISCUSSION TITLE
Using SystemTray.with_title on Ubuntu/Windows

### DISCUSSION THREAD

**Discussion by ekwoka** [tauri-apps/tauri#6302] [Q&A] [answered]
So I get from the docs that the title options on the SystemTray and SystemTrayHandle are only for MacOS.

But how can I set it up so that it only tries to include this for MacOS and ignores it or does nothing on linux and windows?

```
error[E0599]: no method named `set_title` found for struct `SystemTrayHandle` in the current scope
  --> src/main.rs:11:30
   |
11 |     app_handle.tray_handle().set_title(title).unwrap();
   |                                                  ^^^^^^^^^ method not found in `SystemTrayHandle`
```

**Comment by FabianLars** [comment, marked as answer]
Try `#[cfg(target_os = "macos")]`. for example:
```rs
#[cfg(target_os = "macos")]
app_handle.tray_handle().set_title(title).unwrap();
```

**Reply by ekwoka** [reply]
oooh, is there documentation on what this is? I googled a lot and even tried to find something about decorators but couldn't find.

**Reply by ekwoka** [reply]
nvm, I found: https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-attribute

**Reply by FabianLars** [reply]
these are probably good entry points, or at least should show what you need to type into google to find more:
https://doc.rust-lang.org/reference/attributes.html (this is for the `#[]` syntax)
https://doc.rust-lang.org/reference/conditional-compilation.html

Edit: Too slow :D

**Reply by ChaserZ98** [reply]
Tauri v2 is now using predefined predicate `desktop` and `mobile`. Is there any documentation clarifying what exactly they are? Like the corresponding os list of each predicate? I searched a bit but no result.

**Reply by FabianLars** [reply]
desktop is macos linux windows and mobile is android and ios
